e2e/runner.sh

```diff
@@ -79,44 +74,43 @@ echo "Running tests on cluster $MASTER against $REPO."
 echo "Spark images will be created in $IMAGE_REPO"
 
 set -ex
-root=$(pwd)
-
+TEST_ROOT=$(git rev-parse --show-toplevel)
+SPARK_REPO_ROOT="$TEST_ROOT/spark"
 # clone spark distribution if needed.
-if [ -d "spark" ];
+if [ -d "$SPARK_REPO_ROOT" ];
 then
-  (cd spark && git pull origin $BRANCH);
+  (cd $SPARK_REPO_ROOT && git pull origin $BRANCH);
 else
-  git clone $REPO;
+  git clone $REPO $SPARK_REPO_ROOT
 fi
 
-cd spark
+cd $SPARK_REPO_ROOT
 git checkout -B $BRANCH origin/$BRANCH
 ./dev/make-distribution.sh --tgz -Phadoop-2.7 -Pkubernetes -DskipTests
-tag=$(git rev-parse HEAD | cut -c -6)
-echo "Spark distribution built at SHA $tag"
+TAG=$(git rev-parse HEAD | cut -c -6)
+echo "Spark distribution built at SHA $TAG"
 
 if  [[ $DEPLOY_MODE == cloud ]] ;
 then
-  cd dist && ./sbin/build-push-docker-images.sh -r $IMAGE_REPO -t $tag build
+  cd dist && ./sbin/build-push-docker-images.sh -r $IMAGE_REPO -t $TAG build
   if  [[ $IMAGE_REPO == gcr.io* ]] ;
   then
-    gcloud docker -- push $IMAGE_REPO/spark-driver:$tag && \
-    gcloud docker -- push $IMAGE_REPO/spark-executor:$tag && \
-    gcloud docker -- push $IMAGE_REPO/spark-init:$tag
+    gcloud docker -- push $IMAGE_REPO/spark-driver:$TAG && \
+    gcloud docker -- push $IMAGE_REPO/spark-executor:$TAG && \
+    gcloud docker -- push $IMAGE_REPO/spark-init:$TAG
   else
-    ./sbin/build-push-docker-images.sh -r $IMAGE_REPO -t $tag push
+    ./sbin/build-push-docker-images.sh -r $IMAGE_REPO -t $TAG push
   fi
 else
   # -m option for minikube.
-  cd dist && ./sbin/build-push-docker-images.sh -m -r $IMAGE_REPO -t $tag build
+  cd dist && ./sbin/build-push-docker-images.sh -m -r $IMAGE_REPO -t $TAG build
 fi
 
-cd $root/integration-test
-$root/spark/build/mvn clean -Ddownload.plugin.skip=true integration-test \
-          -Dspark-distro-tgz=$root/spark/*.tgz \
+$TEST_ROOT/integration-test/build/mvn clean -Ddownload.plugin.skip=true integration-test \
```

Comment:
Should be  `$TEST_ROOT/spark/build/mvn`

Reply:
or `$SPARK_REPO_ROOT/build/mvn` perhaps

Reply:
Ah that's confusing because we're using the Maven bundled with Spark. We should just provide our own, no?

Reply:
I tried starting with that - but the two mavens I tried were incompatible and caused the spark-build to stall. This works for now and I like that we'll be using the same maven as the spark build, so, moving it in-repo will be easier as well.